### PR TITLE
Fix unused builder for syncing when setting attributes

### LINKED ISSUE #13796: A bug or some dead code in InodeSyncStream.java?
**Alluxio Version:**
lastest master branch

**Describe the bug**
The 607-621 line in InodeSyncStream.java
```Java
SetAttributePOptions.Builder builder = SetAttributePOptions.newBuilder().setMode(new Mode(mode).toProto());
String owner = ufsFpParsed.getTag(Fingerprint.Tag.OWNER);
if (!owner.equals(Fingerprint.UNDERSCORE)) {
    // Only set owner if not empty
    builder.setOwner(owner);
}
String group = ufsFpParsed.getTag(Fingerprint.Tag.GROUP);
if (!group.equals(Fingerprint.UNDERSCORE)) {
     // Only set group if not empty
     builder.setGroup(group);
}
SetAttributeContext ctx = SetAttributeContext.mergeFrom(SetAttributePOptions.newBuilder().setMode(new Mode(mode).toProto())).setUfsFingerprint(ufsFingerprint);
```
The builder is not used after setOwner() and setGroup(), and renew a builder in SetAttributeContext.mergeFrom().

**Urgency**
normal

### PR DESCRIPTION
**Pull request by ZacBlanco**
### What changes are proposed in this pull request?

Utilize the unused SetAttribute builder in the metadat sync code.

I confirmed in a previous incarnation of this code
(https://github.com/Alluxio/alluxio/blame/cc0f1308f4a020cab6099a6622e26a729fe4b755/core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java#L3567-L3580)
That the builder was in-fact being used. Seems like it was lost in
translation when migrating the sync code.

Fixes #13796 

### Why are the changes needed?

Could potentially miss updating the owner on the file.

### Does this PR introduce any user facing changes?

no

### CODE REVIEW

**Comment by apc999** [comment]
possibly to add a UT covering this case?

**Review by ggezer — APPROVED** [review]

File: core/server/master/src/test/java/alluxio/master/file/FileSystemMasterSyncMetadataTest.java [hunk 1]
```diff
@@ -100,23 +106,47 @@ public void after() throws Exception {
...
+
+    FileInfo res = mFileSystemMaster.getFileInfo(uri,
+        GetStatusContext.mergeFrom(GetStatusPOptions.newBuilder().setCommonOptions(
+            FileSystemMasterCommonPOptions.newBuilder().setSyncIntervalMs(0).build())));
+    assertEquals("owner2", res.getOwner());
+    assertEquals("owner2", res.getGroup());
+  }
+
   @Test
   public void listStatusWithSyncMetadataAndEmptyS3Owner() throws Exception {
```
Comment (ggezer) [inline core/server/master/src/test/java/alluxio/master/file/FileSystemMasterSyncMetadataTest.java:146]: This is failing consistently.

**Review by ggezer — CHANGES_REQUESTED** [review]
lgtm after fixing failing UT.

**Review by apc999 — APPROVED** [review]

**Review by ggezer — APPROVED** [review]

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Pull request merged**